### PR TITLE
feat: Add Editor Sidebar and components update route

### PR DESCRIPTION
**Pull request by igorbrasileiro**
## Context
The Live, should handle a way to edit a page component props based on schema, JSON Schema, and update it on storage(supabase).

## Solution
This PR brings a lot of parts, listed bellow
1. Extract the component schemas from components and islands:
The dev.ts script extracts all schemas (JSON schemas) from each component/island module and adds these values to the DecoManifest. ⚠️ Components from subfolders are not listed.
2. Enable the Editor when query string `?editor` is present on request and is private hostname (localhost, deco.page, and deno.dev);
3. When press `ctrl + shift + 'E'` reload the page with `?editor` query string;
4. List current component props to be edited:
The Editor component mounts a form with all components for this page. Can change props, the order of components and remove it
5. Allow adding a new component to the current template;

## Preview
https://github.com/deco-pages/hortifruti/pull/7

### CODE REVIEW

**Comment by igorbrasileiro** [comment]
TODO: change the zod schema to json schema and use react-jsonschema-form

File: utils/zodToObject.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { ZodFirstPartyTypeKind, ZodObject } from "zod";
```
Comment (igorbrasileiro) [inline utils/zodToObject.ts:1]: Remove this in favor of JSON Schema

File: import_map.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
     "twind": "https://esm.sh/twind@0.16.17",
     "twind/": "https://esm.sh/twind@0.16.17/",
     "std/": "https://deno.land/std@0.147.0/",
-    "supabase": "https://esm.sh/@supabase/supabase-js@1.35.4"
+    "supabase": "https://esm.sh/@supabase/supabase-js@1.35.4",
+    "zod": "https://deno.land/x/zod@v3.18.0/mod.ts"
```
Comment (igorbrasileiro) [inline import_map.json:12]: remove

**Review by guifromrio — APPROVED** [review]
A great start!!

**Pull request merged**